### DISCUSSION TITLE
You can pipe emails from mutt

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [dandavison/delta#1141] [General]
Did you know you can pipe emails from [neo]mutt into delta and it highlights diffs as expected?

I don´t  see this mentioned in the README or anything, but it's a very neat selling point IMHO. I was looking for something that did exactly this.

**Comment by dandavison** [comment]
Hi @WhyNotHugo, thanks for pointing that out. That makes sense - delta is intended to print out unchanged all lines that are not diff content, so like a special `cat` that renders embedded diff content. However, the focus of development has been on git/diff output, rather than guaranteeing that it doesn't do something wrong with a non-diff line. Maybe what we should do is make a slightly more formal statement in the docs about what delta is supposed to do.

**Reply by WhyNotHugo** [reply]
Emails are git output; they're the output of `git format-patch`.  :wink: 

Try:

```sh
git format-patch -1 HEAD --stdout | delta
```

**Reply by WhyNotHugo** [reply]
My main point is: the format of the `diff` part of these emails is guaranteed to stay the same as all the rest of the git output, since it needs to remain compatible with thinks like `git am`.

**Reply by dandavison** [reply]
> Emails are git output; they're the output of `git format-patch`

Right, that's true! So it's good that delta is handling this in a reasonable way. The manual has various sections on "Using delta with XXX", would you like to add something for mutt? Incidentally, are the options here limited to piping from mutt into delta, or is there some way to use delta to create a mutt that automatically syntax-highlights diffs? (E.g. if mutt had a hook for preprocessing email bodies before display). That might be attractive for people who use email-based patch workflows?
